### DISCUSSION TITLE
Is this training of the tts_models/multilingual/multi-dataset/your_tts model progressing as designed

### DISCUSSION THREAD

**Discussion by mbarnig** [coqui-ai/TTS#1714] [General Q&A]
I use the recipe multilingual/vits_tts/train_vits_tts.py to train a model with 5 languages (en, fr, pt, de, lb) and 15 voices from scratch, with a customized dataset, assembled from different public sources (LJSpeech, VTCK, SIWIS, mailabs, Edressson Portuguese corpus, Thorsten, Rhasspy and a luxembourgish multilingual corpus). All datasets have been converted to the mailabs-format and the global dataset is balanced between the 5 languages.

I started with a small global dataset with about 6.500 samples. I trained the model up to 70.000 steps, the latest best-model was saved at step 53.801. The related training curves in the tensorboard graphs, shown below, are displayed in orange color.

I was impressed by the first inference results and I expanded the dataset to 12.800 samples, equally segmented for the five languages.
I continued the training with the new dataset by restoring the best-model with 697 /698 layers and with last best loss 15.949144.
The metrics graph seemed to progress as expected, the steps restarted at 53.801.
I stopped the training at step 160.000, the latest best-model was saved at step 133.038.
The related training curves in the tensorboard graphs are displayed in blue color. The inference quality of all voices was sligthly improved.

In a third test I tried to continue training with my expanded dataset by restoring the released model.
I am not sure if this process is called fine-tuningor transfer-learning or somethings else.
I was first worried about the numerous missing layers reported in the logs and about the facts that only 465 / 698 layers were restored and that the step counting restarted at 0. As the training started with a low avg_loss_0 value and as a best_model was saved after the first evaluation, I was happy.

The new training curves are displayed in red color in the tensorboard graphs. I expected to see a decreasing loss, but it increased and finally the loss curves joinded the loss curves from my initial training done from scratch. I stopped the training after step 150.000; the latest best_model was saved at step 122.761. The inference quality is the same as in my second training.

I have now a working multilingual TTS model, including luxembourgish TTS voices, but I don't understand the whole process.

But before asking some questions, I would like to thank the authors of the YourTTS-model for this outstanding project. I would also say thank you to the founders of Coqui.ai for their initiative. And last not least I am grateful to the whole Coqui community for their commitment and help.

Here is the tensorboard graph for evaluation:
[![evaluation](https://www.web3.lu/wp-content/uploads/2022/07/evaluation.jpg)](https://www.web3.lu/wp-content/uploads/2022/07/evaluation.jpg)

Here is the tensorboard graph for training epochs:
[![training](https://www.web3.lu/wp-content/uploads/2022/07/training-epoch.jpg)](https://www.web3.lu/wp-content/uploads/2022/07/training-epoch.jpg)

And here are my questions:

    When restoring my best-model, why is the last layer not restored ? I guess it's because I added 3 characters in the configuration list.
    Why are 233 layers missing when restoring the released model ?
    Why is the loss increasing when training with the restored released model ?
    What is the name of the process to train a pretrained model with existing and additional languages and voices ?
    Are the tensorboard curves progressing as expected ?

**Comment by hithesh-sankararaman** [comment]
After training your model.
How did u test it ?
What is the command ? What all needs to be changed in config.json ?
